sabelotodo/models.py

```diff
@@ -0,0 +1,29 @@
+from sqlalchemy import Column, Integer, String, Boolean, ForeignKey, UniqueConstraint
+from sqlalchemy.orm import relationship
+from sabelotodo.database import Base
+
+class Item(Base):
+
+    # A to-do item. Items can form trees, with each storing the id of its
+    # parent if it has one. The order field represents the order among
+    # siblings, and in any set of siblings, order numbers must be unique.
+
+    __tablename__ = 'items'
+    __table_args__ = (UniqueConstraint('order','parent_id',name='sibling_order'),)
+    id = Column(Integer, primary_key=True)
+    name = Column(String(256))
+    order = Column(Integer, unique=True)
+    done = Column(Boolean)
+    description = Column(String)
+    parent_id = Column(Integer, ForeignKey('items.id'))
+
+    def __init__(self, name, order=None, 
+            done=False, description="", parent_id=None):
+        self.name = name
```

Comment:
I feel like there might be a nice way to do this whole deal in python 3.8, but I'm not sure.

Reply:
The one I know of is the attrs package, which is pretty nifty. Worth switching?

Reply:
I think there's something native now. Not worth changing it atm, though.

Reply:
Ah, dataclass looks like it. Looking for stuff online about dataclasses and Flask, I'm finding a lot of "I wish this worked" and "Why won't it work?" and no clear and unambiguous "Here's how to make it work." So yeah, let's skip it.